### DISCUSSION TITLE
curl --dry-run ?

### DISCUSSION THREAD

**Discussion by tokudan** [curl/curl#12953] [Ideas]
Hello,

I'm looking for a sort of "dry-run" option for the curl binary where it processes all options but stops just short of actually making network connections.

The background for my question:
I'm abusing curl in a bash script to build a URI with proper escaping of variables with URL encoding. Those variables are a great feature, btw.
However, I do not need curl to make the connection, I need to present the URI to the user and ask them to login at that URI in their own browser as part of an oauth2 process.
I figured out I can trick curl into avoiding the actual connection by restricting the protocol like this:
`$ curl --proto http -w '%{url}' --variable a=b 'https://example.com/{{a:url}}'`
However this feels like an abuse of a security feature and I'm not sure how reliable this will be in the future.

**Comment by dfandrich** [comment]
The whole idea seems like a bit of an abuse of curl. Are you sure you don't want to be using trurl for this?

**Reply by emanuele6** [reply]
I.E.
```console
$ trurl -s 'path=b?+&#x' https://example.com
https://example.com/b%3f%2b%26%23x
```

The only difference would be that if your variable contains a `/`,  `trurl -s 'path=....'` will keep it as a `/`, while `curl --expand-url https://example.org/{{a:url}}` will encode it as `%2f`; but that is probably what you want anyway:
```console
$ trurl -s 'path=b/?+&#x' https://example.com
https://example.com/b/%3f%2b%26%23x
$ curl --variable 'a=b/?+&#x' -w%{url}\\n -gIsSo/dev/null --expand-url https://example.com/{{a:url}}
https://example.com/b%2F%3F%2B%26%23x
```
But if you really want that behaviour, you can also have it with `trurl` using `-s 'path=/' -a 'path=....'` instead of just `-s 'path=....'`:
```console
$ trurl -s 'path=/' -a 'path=b/?+&#x' https://example.com
https://example.com/b%2f%3f%2b%26%23x
```

**Reply by tokudan** [reply]
I wasn't aware of the existence of trurl, on first look this looks exactly what I need.
Only downside is, it's less likely to be installed on systems than curl, but I can live with that for my use case.
Thanks!
